### PR TITLE
Fix and improve our FactoryGirl factories.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

File: dashboard/test/factories.rb [hunk 1]
```diff
@@ -28,10 +30,12 @@
       type "SingleUserExperiment"
     end
   end
+
   factory :section_hidden_stage do
-    section nil
-    stage nil
```
Comment (ashercodeorg) [inline dashboard/test/factories.rb:33]: These fields are enforced to be non-NULL.
```
SHOW CREATE TABLE section_hidden_stages;
+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table                 | Create Table                                                                                                                                                                                                                                                                                                                                                                          |
+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| section_hidden_stages | CREATE TABLE `section_hidden_stages` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `section_id` int(11) NOT NULL,
  `stage_id` int(11) NOT NULL,
  PRIMARY KEY (`id`),
  KEY `index_section_hidden_stages_on_section_id` (`section_id`),
  KEY `index_section_hidden_stages_on_stage_id` (`stage_id`)
) ENGINE=InnoDB AUTO_INCREMENT=24635 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci |
+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

File: dashboard/test/factories.rb [hunk 2]
```diff
@@ -118,23 +122,21 @@
...
+        end
       end
-    end
 
-    factory :old_student do
-      user_type User::TYPE_STUDENT
-      birthday Time.zone.today - 30.years
+      factory :old_student do
+        birthday Time.zone.today - 30.years
+      end
```
Comment (Bjvanminnen) [inline dashboard/test/factories.rb:139]: 30 years is old 😢

File: dashboard/test/factories.rb [hunk 3]
```diff
@@ -151,8 +153,8 @@
   end
 
   factory :districts_users do
-    district nil
-    user nil
```
Comment (ashercodeorg) [inline dashboard/test/factories.rb:155]: These fields are enforced to be non-NULL.
```
SHOW CREATE TABLE districts_users;
+-----------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table           | Create Table                                                                                                                                                                                                                                                                                                                         |
+-----------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| districts_users | CREATE TABLE `districts_users` (
  `user_id` int(11) NOT NULL,
  `district_id` int(11) NOT NULL,
  KEY `index_districts_users_on_user_id_and_district_id` (`user_id`,`district_id`),
  KEY `index_districts_users_on_district_id_and_user_id` (`district_id`,`user_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci |
+-----------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

File: dashboard/test/factories.rb (hunk 2 above)
Comment (ashercodeorg) [reply dashboard/test/factories.rb:139]: You calling me older than old? 😭

**Pull request merged**